provision/acc_provision/acc_provision.py

```diff
@@ -950,11 +931,88 @@ def provision(args, apic_file, no_random):
     return ret
 
 
+DEFAULT_FLAVORS_PATH = os.path.dirname(os.path.realpath(__file__)) + "/flavors.yaml"
+VERSION_FIELDS = [
+    "cnideploy_version",
+    "aci_containers_host_version",
+    "opflex_agent_version",
+    "aci_containers_controller_version",
+    "openvswitch_version",
+]
+
+
+def check_yaml(text_to_verify):
+    try:
+        flavors_yaml = yaml.load(text_to_verify)
+        if 'default_flavor' in flavors_yaml:
+            return flavors_yaml
+    except yaml.YAMLError as exc:
+        print(exc)
+        raise Exception
+
+
+def default_flavors_path():
+    with open(DEFAULT_FLAVORS_PATH, 'r') as res:
+        info("Loading flavors from default flavors.yaml: " + DEFAULT_FLAVORS_PATH)
+        valid_yaml = check_yaml(res)
+        if valid_yaml:
+            set_flavors(valid_yaml)
+    return
+
+
+def get_flavors(flavors_url):
+    if flavors_url == 'blah':
+        return default_flavors_path()
+    flag = False
+    try:
+        # try as a URL
+        res = requests.get(flavors_url)
+        if res and res.status_code == 200:
+            valid_yaml = check_yaml(res.text)
+            if valid_yaml:
+                info("Loading flavors from URL: " + flavors_url)
+                set_flavors(valid_yaml)
+                flag = True
+        else:
+            raise Exception
+    except Exception:
+        # try as a local file
+        if exists(flavors_url):
+            with open(flavors_url, 'r') as res:
+                valid_yaml = check_yaml(res)
+                if valid_yaml:
+                    info("Loading flavors from local file: " + flavors_url)
+                    set_flavors(valid_yaml)
+                    flag = True
+    finally:
+        # try the static file in repo
+        if not flag:
+            return default_flavors_path()
+
+
+def set_flavors(doc):
+    global DEFAULT_FLAVOR, VERSIONS, DEFAULT_FLAVOR_OPTIONS, CfFlavorOptions, FLAVORS
+    DEFAULT_FLAVOR = doc['default_flavor']
+    VERSIONS = doc['versions']
+    DEFAULT_FLAVOR_OPTIONS = doc['kubeFlavorOptions']
+    CfFlavorOptions = doc['cfFlavorOptions']
+    CfFlavorOptions['template_generator'] = generate_cf_yaml
+    FLAVORS = doc['flavors']
+
+
 def main(args=None, apic_file=None, no_random=False):
     # apic_file and no_random are used by the test functions
     if args is None:
         args = parse_args()
 
+    config = config_user(args.config)
+    if config:
+        if 'flavors_url' not in config:
+            config['flavors_url'] = {}
+            config['flavors_url']['path'] = "blah"
```

Comment:
Assigned a specific(absurd) value to check for it later in the code, so that we know the code set it. 
In this case I guess it shouldn't matter because we anyway want to default to the flavors.yaml in our path. I could change it to None as well.